scripts/wsmux.sh

```diff
@@ -99,30 +105,97 @@ wsmux() {
 						if [[ -f ~/$(echo $BINDING | cut -d ':' -f2)/devel/setup.sh ]]; then
-							SELECTION=~/$(echo $BINDING | cut -d ':' -f2)
+							SELECTION=$(echo $BINDING | cut -d ':' -f2)
 
 							# Source the workspace runcom file if it exists
-							if [[ -f $SELECTION/.wsrc ]]; then
-								source $SELECTION/.wsrc
+							if [[ -f ~/$SELECTION/.wsrc ]]; then
+								source ~/$SELECTION/.wsrc
 							fi
 
 						# Otherwise, prompt the user to check themself
 						else
-							echo "The specified workspace is not valid. I don't trust like that..."
-							echo "Please specify the name of a workspace in the home directory"
+							echo "The bound workspace is not valid. I don't trust like that..."
+							echo "Please bind to a workspace in the home directory"
 							SELECTION="false"
+							shift $#
 						fi
 					fi
 				done
 				shift 1
 				;;
+			-h|--help)
+				echo "Usage: wsmux [OPTION]... [WORKSPACE]"
+				echo "Quick catkin workspace switcher."
+				echo ""
+				echo "Option		GNU long option		Meaning"
+				echo "-b [WORKSPACE]	--bind			Bind an SSH client to a workspace"
+				echo "-c		--connect		Select the SSH client's workspace"
+				echo "-h		--help			Display the help menu"
+				echo "-l		--list			List the detected workspaces"
+				echo "-p [WORKSPACE]	--pull			Pull the latest changes from upstream"
+				echo "-s		--show			Show the selected workspace"
+				echo "-u		--unbind		Remove the binding for an SSH client"
+				SELECTION="false"
+				shift 1
+				;;
 			-l|--list)
 				if [[ ! -z "COMPREPLY" ]]; then
 					echo "${COMPREPLY[@]}" | sed 's/ /  /g'
 				fi
 				SELECTION="false"
 				shift 1
 				;;
+			-p|--pull)
+				SELECTION=""
+
+				# Make sure the user understands what they are doing
+				echo -n "Are you sure you want to delete changes on repository master branches? [y/N] " && read SELECTION
+				if [[ "$SELECTION" != "Y" && "$SELECTION" != "y" ]]; then
+					SELECTION="false"
+					shift $#
+
+				# If a workspace name was entered, select that workspace
+				elif [[ -f ~/$2/devel/setup.sh ]]; then
+					SELECTION=$2
+					shift 2
+
+				# If no workspace name was entered, use the currently selected workspace
+				elif [[ -z "$2" && -f ~/$(echo $CATKIN_DIR | rev | cut -d '/' -f1 | rev)/devel/setup.sh ]]; then
+					SELECTION=$(echo $CATKIN_DIR | rev | cut -d '/' -f1 | rev)
+					shift 1
+
+				# Prompt the user to check themself if the selected workspace is not valid
+				else
+					echo "The specified workspace is not valid. I don't trust like that..."
+					echo "Please specify the name of a workspace in the home directory"
+					SELECTION="false"
+					shift $#
+				fi
+
+				if [[ "$SELECTION" != "false" ]]; then
+					for FILE in ~/$SELECTION/src/*; do
+
+						# Verify that the directory is a Git repository
+						if [[ -d $FILE/.git ]]; then
+							cd $FILE
+
+							# Verify that the repository's working directory is clean
+							if $(git diff-index --quiet HEAD --); then
```

Comment:
So is this checking for the exit code? Looks like this command returns exit code 1 for an unclean working directory. So isn't this the opposite of what you want (assuming if runs for positive exit codes)